### PR TITLE
Show vote validate inputs #39

### PR DESCRIPTION
**Pull request by valentinbujdoso**

### CODE REVIEW

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+		assertThrows(
+			() -> voteManager.showVote(adminInfo.voteId, invalidAdminKey)
+		).assertMessageIs(String.format("Illegal adminKey: %s",invalidAdminKey));
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void proper_voteId_and_adminKey_check_name() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: please use a name useable in the documentation. It looks silly right now:
https://adadocs.demokracia.rulez.org/PDEngine/edemo/feature.show.vote.validate.inputs.39/engine-implementedBehaviours.html#returns%20the%20vote%20in%20json

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+	public void invalid_adminKey_is_rejected() throws ReportedException {
+		String invalidAdminKey = RandomUtils.createRandomKey();
+		assertThrows(
+			() -> voteManager.showVote(adminInfo.voteId, invalidAdminKey)
+		).assertMessageIs(String.format("Illegal adminKey: %s",invalidAdminKey));
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: Please start a new test class for a new behaviour, as both of them contain more test cases.
See https://adadocs.demokracia.rulez.org/zenta-tools/magwas/master/zenta-tools/codingrules.html#codereq15
"Use classes with only a few methods"

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("name"), vote.name);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void proper_voteId_and_adminKey_check_canAddIn() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("canAddIn"), vote.canAddin);
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void proper_voteId_and_adminKey_check_creationTime() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("creationTime"), vote.creationTime);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void proper_voteId_and_adminKey_check_choices() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+		JSONObject jsonObject = new JSONObject(vote.choices);
+
+		assertEquals(result.get("choices").toString(), jsonObject.toString());
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void proper_voteId_and_adminKey_check_canEndorse() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("canEndorse"), vote.canEndorse);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void proper_voteId_and_adminKey_check_countedAssurances() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+		JSONArray jsonArray = new JSONArray(vote.countedAssurances);
+		
+		assertEquals(result.get("countedAssurances").toString(), jsonArray.toString());
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void proper_voteId_and_adminKey_check_neededAssurances() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+		
+		
+		assertEquals(result.get("neededAssurances").toString(), jsonArray.toString());
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void proper_voteId_and_adminKey_check_minEndorsements() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("minEndorsements"), vote.minEndorsements);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void proper_voteId_and_adminKey_check_id() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("id"), vote.id);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void proper_voteId_and_adminKey_check_canView() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("canView"), vote.canView);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void proper_voteId_and_adminKey_check_canVote() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above

**Review by magwas — CHANGES_REQUESTED** [review]

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java (hunk 1 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java (hunk 2 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java (hunk 3 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java (hunk 4 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java (hunk 5 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java (hunk 8 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java (hunk 9 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java (hunk 10 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java (hunk 11 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java (hunk 12 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java (hunk 2 above)
Comment (magwas) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: it is still in this class

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+		assertThrows(
+			() -> voteManager.showVote(adminInfo.voteId, invalidAdminKey)
+		).assertMessageIs(String.format("Illegal adminKey: %s",invalidAdminKey));
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_name() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: this is still not a proper sentence, like "the name attribute contains the name of the vote"

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 14]
```diff
@@ -0,0 +1,169 @@
...
+		
+		assertEquals(result.get("name"), vote.name);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_canAddIn() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 15]
```diff
@@ -0,0 +1,169 @@
...
+		
+		assertEquals(result.get("canAddIn"), vote.canAddin);
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_creationTime() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 16]
```diff
@@ -0,0 +1,169 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("creationTime"), vote.creationTime);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_choices() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 17]
```diff
@@ -0,0 +1,169 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		JSONObject jsonObject = new JSONObject(vote.choices);
+
+		assertEquals(result.get("choices").toString(), jsonObject.toString());
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: I am not convinced this properly serializes a non-empty set of choices.

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 18]
```diff
@@ -0,0 +1,169 @@
...
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_canEndorse() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above about the names

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 19]
```diff
@@ -0,0 +1,169 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("canEndorse"), vote.canEndorse);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_countedAssurances() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above about the names, and also some (although subminimal) concerns about serialisation here.

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 20]
```diff
@@ -0,0 +1,169 @@
...
+		JSONArray jsonArray = new JSONArray(vote.countedAssurances);
+		
+		assertEquals(result.get("countedAssurances").toString(), jsonArray.toString());
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_neededAssurances() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 21]
```diff
@@ -0,0 +1,169 @@
...
+		
+		
+		assertEquals(result.get("neededAssurances").toString(), jsonArray.toString());
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_minEndorsements() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above about the naming

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 22]
```diff
@@ -0,0 +1,169 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("minEndorsements"), vote.minEndorsements);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_id() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above about the naming

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 23]
```diff
@@ -0,0 +1,169 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("id"), vote.id);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_canView() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above about the naming

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java [hunk 24]
```diff
@@ -0,0 +1,169 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("canView"), vote.canView);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_canVote() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: see above about the naming

**Review by magwas — CHANGES_REQUESTED** [review]
please read through the coding rules once more.
test case names becomepart of the documentation, so it is important that they are understandable by those whom the documentation targets
Also it is not inherently obvious that a map of strings and choices is properly serializes into json without further effort.

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java (hunk 2 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: Sorry I missed it. Now I fixed.

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java (hunk 13 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java (hunk 17 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: Ok I will check it.

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java [hunk 25]
```diff
@@ -0,0 +1,147 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("name"), vote.name);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_canAddIn_attribute_contains_the_addability_of_the_vote() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java]: the canAddin attribute tells whether voters can add choices to the vote at the moment

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java [hunk 26]
```diff
@@ -0,0 +1,147 @@
...
+		JSONObject jsonObject = new JSONObject(vote.choices);
+
+		assertEquals(result.get("choices").toString(), jsonObject.toString());
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_canEndorse_attribute_contains_the_endorsable_of_the_vote() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java]: the canEndorse attribute tells whether voters endorse choices of the vote at the moment

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java [hunk 27]
```diff
@@ -0,0 +1,147 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("id"), vote.id);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_canView_attribute_contains_the_viewability_of_the_vote() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java]: the canView attribute tells whether voters can view the results

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java [hunk 28]
```diff
@@ -0,0 +1,147 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("canView"), vote.canView);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_canVote_attribute_contains_the_voteability_of_the_vote() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java]: the canVote attribute tells whether votes can be cast at the moment

**Review by magwas — CHANGES_REQUESTED** [review]
I was hesitating whether to approve or request changes.
But the quality of the documentation is at stake.

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java (hunk 28 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java (hunk 27 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java (hunk 26 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java (hunk 25 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java (hunk 17 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: Ok I checked it, I have to create a toJson method for the Choices too.
The question is that do we need to test these methodes too?

Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java [hunk 29]
```diff
@@ -0,0 +1,150 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("name"), vote.name);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_canAddIn_attribute_contains_the_voters_can_add_choices_to_the_vote() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java]: at least a "whether" is missing here.
tha canAddin attribute contains whether the voters can add choices to the vote

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java [hunk 30]
```diff
@@ -0,0 +1,150 @@
...
+		JSONArray jsonArray = vote.createChoicesJson(vote.choices);
+		
+		assertEquals(result.getJSONArray("choices").toString(), jsonArray.toString());
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_canEndorse_attribute_contains_the_voters_endorse_choices_of_the_vote() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java]: the canEndorse attribute contains whether the voters can endorse choices of the vote

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java [hunk 31]
```diff
@@ -0,0 +1,150 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("id"), vote.id);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_canView_attribute_contains_the_voters_can_view_the_results() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java]: the canViev attribute contains whether the voters can view the results

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java [hunk 32]
```diff
@@ -0,0 +1,150 @@
...
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("canView"), vote.canView);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void the_canVote_attribute_contains_the_votes_can_be_cast() throws ReportedException {
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java]: the canVote attribute contains whether the votes can be cast

**Review by magwas — CHANGES_REQUESTED** [review]
Grammatik macht frei!
(Sorry about being grammar nazi, but it does go to the documentation.)

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java (hunk 30 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java (hunk 32 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java (hunk 29 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java (hunk 31 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java (hunk 29 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java]: No problem. I would like to be free. :) 
And I would like to give you high quality code and documentation.

**Review by magwas — APPROVED** [review]

**Comment by magwas** [comment]
testcaseCount=13

**Pull request merged**